crates/modelardb_compression_python/src/lib.rs

```diff
@@ -0,0 +1,196 @@
+#[pymodule]
+fn modelardb_compression_python(_python: Python<'_>, py_module: &PyModule) -> PyResult<()> {
+    py_module.add_function(wrap_pyfunction!(compress, py_module)?)?;
+    py_module.add_function(wrap_pyfunction!(decompress, py_module)?)?;
+
+    Ok(())
+}
+
+/// Compress a [`RecordBatch`] containing an univariate time series within `error_bound`. An
+/// `univariate_id` can optionally be set to specify which univariate time series the compressed
+/// segments was produced from, if none is set zero is used. An [`ArrowException`] is returned if
+/// the schema of `uncompressed` does not match [`UNCOMPRESSED_SCHEMA`] or if [`pyo3`] cannot
+/// convert the arguments from Python to Rust and the result from Rust to Python.
+#[pyfunction(univariate_id = 0)]
+fn compress<'a>(
+    uncompressed: &'a PyAny,
+    error_bound: &'a PyAny,
+    univariate_id: u64,
+    python: Python<'a>,
+) -> PyResult<&'a PyAny> {
+    // Convert the record batch from Python to Rust and validate the arguments.
+    let uncompressed = convert::python_record_batch_to_rust(uncompressed)?;
+    if !fields_has_equal_types(&uncompressed.schema(), &UNCOMPRESSED_SCHEMA.0) {
+        return Err(ArrowException::new_err(
+            "Uncompressed can only contain millisecond timestamps and float32 values.".to_owned(),
+        ));
+    }
+
+    let error_bound = models::ErrorBound::try_new(error_bound.extract::<f32>()?)
+        .map_err(|error| convert::to_py_err(ArrowError::InvalidArgumentError(error.to_string())))?;
+
+    // Compress the timestamps and values to segments containing metadata and models.
+    let timestamps = modelardb_common::array!(uncompressed, 0, TimestampArray);
+    let values = modelardb_common::array!(uncompressed, 1, ValueArray);
+
+    // unwrap() is safe as the timestamps and values are guaranteed to be the same length.
+    let compressed = modelardb_compression::try_compress(
+        univariate_id,
+        timestamps,
+        values,
+        error_bound,
+        &COMPRESSED_SCHEMA,
+    )
+    .unwrap();
+
+    // Convert and return the finished compressed PyArrow record batch.
+    convert::rust_record_batch_to_python(compressed, python)
+}
+
+/// Decompress a [`RecordBatch`] containing compressed segments to one more univariate time series.
+/// An [`ArrowException`] is returned if the schema of `compressed` does not match
+/// [`COMPRESSED_SCHEMA`] or if [`pyo3`] cannot convert the argument from Python to Rust and the
+/// result from Rust to Python.
+#[pyfunction]
+fn decompress<'a>(compressed: &'a PyAny, python: Python<'a>) -> PyResult<&'a PyAny> {
+    // Convert the record batch from Python to Rust and validate the arguments.
+    let compressed = convert::python_record_batch_to_rust(compressed)?;
+    if !fields_has_equal_types(&compressed.schema(), &COMPRESSED_SCHEMA.0) {
+        let error = ArrowError::InvalidArgumentError(
+            "Compressed must only contain compressed segments with metadata and models.".to_owned(),
```

Comment:
Maybe change to `Compressed data must...` instead. Maybe also change `must` to `can` to be consistent with the error message in `compress()`.

Reply:
I changed `must` to `can` for consistency, but maybe keep `compressed` and `uncompressed` without `data` to match name of the arguments?